core/market/decentralized/src/db/dao/offer.rs

```diff
@@ -58,6 +92,25 @@ impl<'c> OfferDao<'c> {
         .await
     }
 
+    pub async fn checked_insert(
```

Comment:
All inserts should be checked. No need to highlight this in name

Reply:
initial should not I believe...

Reply:
Probably we should have only one function. Caller decides if this failing check is an error or valid situation.
But doesn't matter. Let's merge this quickly

Reply:
Ok, now there is a single insert function which returns resulting Offer with `insertion_ts`.